File: Search/Search2DMatrixII/Search2DMatrixII.py
class Solution:
    def search2DMatrixII(self, matrix, target):
        if matrix is none or matrix[0] is none:
            return

        occur = 0
        row, col = 0, len(matrix[0]) - 1

        while row < len(matrix) and col >= 0:
            if matrix[row][col] == target:
                col -= 1
                occur += 1
            elif matrix[row][col] < target:
                row += 1
            else:
                col -= 1

        return occur

File: Search/Search2DMatrixII/Search2DMatrixII.cc
class Solution {
    public:
        int search2DMatrixII(vector<vector<int>> &matrix, int target) {
            if(matrix == null || matrix.size() == 0 || matrix[0].size() == 0)
                return -1;

            const int ROW = matrix.size();
            const int COL = matrix[0].size();

            int row = 0;
            int col = COL - 1;

            while(row < ROW && col >= 0) {
                if(matrix[row][col] == target) {
                    occur++;
                    col--;
                } else if(matrix[row][col] > target) {
                    col--;
                } else
                    row++;
            }
            return occur;
        }
};
